actions/views/channel.js

```diff
@@ -100,20 +102,32 @@ export function leaveChannel(channelId) {
         const state = getState();
         const myPreferences = getMyPreferences(state);
         const currentUserId = getCurrentUserId(state);
-        const currentTeamId = getCurrentTeamId(state);
+        const currentTeam = getCurrentTeam(state);
 
         if (isFavoriteChannel(myPreferences, channelId)) {
             dispatch(unfavoriteChannel(channelId));
         }
 
         const teamUrl = getCurrentRelativeTeamUrl(state);
-        LocalStorageStore.removePreviousChannelName(currentUserId, currentTeamId);
-        browserHistory.push(teamUrl);
-
+        LocalStorageStore.removePreviousChannelName(currentUserId, currentTeam.id);
         const {error} = await dispatch(leaveChannelRedux(channelId));
         if (error) {
             return {error};
         }
+        const prevChannelName = LocalStorageStore.getPreviousChannelName(currentUserId, currentTeam.id);
+        const channelsInTeam = getChannelsNameMapInCurrentTeam(state);
+        const prevChannel = getChannelByName(channelsInTeam, prevChannelName);
+        if (!prevChannel || !getMyChannelMemberships(getState())[prevChannel.id]) {
+            LocalStorageStore.removePreviousChannelName(currentUserId, currentTeam.id);
+        }
+        if (getMyChannels(getState()).filter((c) => c.type === 'O' || c.type === 'P').length === 0) {
```

Comment:
Probably better to use Constants.OPEN_CHANNEL/PRIVATE_CHANNEL here

Reply:
Done